source/digits_hits/src/GateRestrictedVolume.cc

```diff
@@ -0,0 +1,67 @@
+/** ----------------------
+  Copyright (C): OpenGATE Collaboration
+  This software is distributed under the terms
+  of the GNU Lesser General  Public Licence (LGPL)
+  See LICENSE.md for further details
+  ----------------------*/
+
+#include "GateRestrictedVolume.hh"
+
+#include "G4UnitsTable.hh"
+
+#include "GateRestrictedVolumeMessenger.hh"
+#include "GateTools.hh"
+
+GateRestrictedVolume::GateRestrictedVolume(GatePulseProcessorChain* itsChain,
+                                       const G4String& itsName)
+  : GateVPulseProcessor(itsChain, itsName)
+{
+  m_messenger = new GateRestrictedVolumeMessenger(this);
```

Comment:
if you use std::unique_ptr you dont have to call:
`m_messenger = new GateRestrictedVolumeMessenger(this);`

Reply:
Is it really need to keep the m_messenger? In fact, It is not used anywhere in this class... I copied this implementation from other class but as far as I see it is really not used anywhere... Could you please look at that?

Reply:
I delete it and GATE is compiling and runinng without any error and warning. The results also looks fine.

Reply:
Sorry, I made a mistake it doesn't working properly...